requirements.txt

```diff
@@ -1,11 +1,11 @@
 bayesian-changepoint-detection==0.2.dev1
 changepy==0.3.1
-hyperopt==0.1.2
-numpy==1.17.0
-pandas==0.25.1
+hyperopt==0.2.5
```

Comment:
@sayanchk any concerns with a major version change here?

Reply:
@snazzyfox We should test the suggested changes if it's breaking anything or treating anything differently compared to the older versions. I am aware of scikit-learn to break the build with newer versions.

I think we should revisit all the dependency versions and should support wider ranges of recent version instead of depending on any specific version. That will require rigorous testing all the dependencies with the latest versions.

Reply:
@sayanchk any pointers how improve this PR? Currently i just rely on existing tests to check if bump successful.